src/v/storage/disk_log_impl.h

```diff
@@ -379,7 +407,12 @@ class disk_log_impl final : public log {
     size_t _suffix_truncation_indicator{0};
 
     std::optional<model::offset> _cloud_gc_offset;
+
+    // The offset at which the last window compaction finished, above which keys
+    // have been fully deduplicated. The next round of window compaction
```

Comment:
> The next round of window compaction will start from this offset.

Maybe: The next round of window compaction can skip segments above this offset if no new segments have been created since last window compaction.
This is how I read https://github.com/redpanda-data/redpanda/blob/6fe0605a473e8b0afd218904b7208ef11efd2f18/src/v/storage/disk_log_impl.cc#L533